### ISSUE TITLE
Router issue

### ISSUE THREAD

**Issue by tib** [hummingbird-project/hummingbird#478]
Hi,

I believe I discovered a router issue in HB2beta, given the following router setup:

```swift
import Hummingbird

func buildApplication(
    configuration: ApplicationConfiguration
) -> some ApplicationProtocol {
    let router = Router()
    
    router.get("/foo/bar") { _, _ in "foo-bar"}
    router.get("/foo/{id}/baz") { _, _ in "foo-bar-baz" }

    let app = Application(
        router: router,
        configuration: configuration
    )
    return app
}
```

Using this test case:

```swift
@testable import App
import Hummingbird
import HummingbirdTesting
import XCTest

final class AppTests: XCTestCase {

    func testApp() async throws {
        let app = buildApplication(
            configuration: .init()
        )
        try await app.test(.live) { client in
            
            try await client.execute(
                uri: "/foo/bar/baz",
                method: .get
            ) { response in
                XCTAssertEqual(response.status, .ok)
                XCTAssertEqual(String(buffer: response.body), "foo-bar-baz")
            }
        }
    }
}
```

The test fails with a 404 response, but I'm expecting the router to resolve a route for the `foo/bar/baz` request and bar should be the value of the `{id}`. 

Is this a real issue or am I doing something wrong? 

Thank you for the help.
Tib

**Comment by adam-fowler** [comment]
Looks like the rewinding of the router path stack isn't working. It is an issue, we'll look into it.

**Comment by adam-fowler** [comment]
@joannis has fixed this in #479
